include/ensmallen_bits/pso/update_policies/lbest_update.hpp

```diff
@@ -0,0 +1,163 @@
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_PSO_UPDATE_POLICIES_LBEST_UPDATE_HPP
+#define ENSMALLEN_PSO_UPDATE_POLICIES_LBEST_UPDATE_HPP
+#include <assert.h>
+
+namespace ens {
+
+/**
+ * The local best version (lbest) of PSO in which particles communicate with
+ * only two neighbours each, thus forming a ring topology amongst them. This
+ * approach allows PSO to converge at the global minimum, but takes
+ * significantly more iterations to do so.
+ *
+ * The lbest update scheme is described as follows:
+ *
+ * \f{eqation}{
+ * v_{i+1} = \phi (v_i + c_1 * r_1 * (p_{best} - p_{current}) +
+ *           c_1 * r_1 * (l_{best} - p_{current}))
+ * \f}
+ *
+ * where \f$ v_i \f$ is the velocity of a particle in iteration \f$ i \f$,
+ *       \f$ p_{best} \f$ is the best position of an individual particle,
+ *       \f$ p_{current} \f$ is the current position of the particle,
+ *       \f$ l_{best} \f$ is the local best position,
+ *       \f$ r_1 \f$ and \f$ r_2 \f$  are standard uniform random variables,
+ *       \f$ c_1 \f$ is the exploitation factor,
+ *       \f$ c_2 \f$ is the exploration factor, and
+ *       \f$ \phi \f$ is the constriction factor.
+ *
+ * For more information, refer the following:
+ *
+ * @code
+ * @article{Poli2007,
+ *          doi = {10.1007/s11721-007-0002-0},
+ *          url = {https://doi.org/10.1007/s11721-007-0002-0},
+ *          year = {2007},
+ *          month = aug,
+ *          publisher = {Springer Science and Business Media {LLC}},
+ *          volume = {1},
+ *          number = {1},
+ *          pages = {33--57},
+ *          author = {Riccardo Poli and James Kennedy and Tim Blackwell},
+ *          title = {Particle swarm optimization},
+ *          journal = {Swarm Intelligence}
+ * }
+ * @endcod
+ */
+class LBestUpdate
+{
+ public:
+  /**
+   * The Initialize method is called by PSO Optimizer method before the start of
+   * the iteration process. It calculates the value of the constriction
+   * coefficent, initializes the local best indices of each particle to itself,
+   * and sets the shape of the r1 and r2 vectors.
+   *
+   * @param exploitationFactor Influence of personal best achieved.
+   * @param explorationFactor Influence of neighbouring particles.
+   * @param numParticles The number of particles in the swarm.
+   * @param iterate The user input, used for shaping intermediate vectors.
+   */
+  void Initialize(const double& exploitationFactor,
+                  const double& explorationFactor,
+                  const size_t& numParticles,
+                  const arma::mat& iterate)
+  {
+    // Copy values to aliases.
+    n = numParticles;
+    c1 = exploitationFactor;
+    c2 = explorationFactor;
+
+    // Calculate the constriction factor
+    static double phi = c1 + c2;
+    assert(phi > 4.0 && "The sum of the exploitation and exploration factors "
+                        "must be greater than 4.");
```

Comment:
Indentation should be 4 spaces, if this is a new line.

Reply:
But it is a part of the previous line of code. That's the reason I aligned it with the text. I'll change it if required.

Reply:
Missed the comment; I think we should follow the style as used here:

https://github.com/mlpack/mlpack/blob/2326bf8bd0b380c17eed6b5b1c34fbfab1563774/src/mlpack/tests/sfinae_test.cpp#L102-L108